cns/service/main.go

```diff
@@ -360,7 +360,7 @@ type NodeInterrogator interface {
 }
 
 // RegisterNode - Tries to register node with DNC when CNS is started in managed DNC mode
-func registerNode(httpc *http.Client, httpRestService cns.HTTPService, dncEP, infraVnet, nodeID string, ni NodeInterrogator) error {
+func registerNode(ctx context.Context, httpClient acn.HTTPClient, httpRestService cns.HTTPService, dncEP, infraVnet, nodeID string, ni NodeInterrogator) error {
```

Comment:
Instead of a global `acn.HTTPClient` interface, these methods can use a local (private) interface. Remember that in Go, anything that has the necessary methods satisfies the interface. Any type is acceptable that provides the behavior that we need, _including existing types_ like the stdlib `http.Client`

```suggestion
type httpDoer interface {
	Do(req *http.Request) (*http.Response, error)
}

func registerNode(ctx context.Context, httpClient httpDoer, httpRestService cns.HTTPService, dncEP, infraVnet, nodeID string, ni NodeInterrogator) error {
```

Reply:
Thanks for the tip. Addressed.